lessons/git/intro/slides-2017.Rmd

```diff
@@ -0,0 +1,305 @@
+---
+title: "Intro to Git"
+author: "Shyam Srinivasan"
+date: "October 5, 2017"
+output: beamer_presentation
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = FALSE)
+```
+
+# Learning expectations
+
+This won't even touch the surface of git... But! Hopefully, you:
+
+- Recognize the power of using version control
+- Know the basic tools to get started using git
+- Where to go for help
+
+But! No expectation to actually start using it :)
+
+
+# Getting started with Git and GitHub
+
+Installing and running Git and GitHub
+
+## Linux
+https://www.howtoforge.com/tutorial/install-git-and-github-on-ubuntu-14.04/
+
+* Dowload and Install Git for Linux
+`sudo apt-get install git`
+
+## Windows/MacOS
+https://desktop.github.com/
+
+* Make sure you install posh-git along with GitHub
+
+# Configuring your setup:
+
+```bash
+git config --global user.name "Your Name"
+git config --global user.email "you@some.domain"
+```
+
+Optional Configurations
+```bash
+git config --global color.ui "auto" 
+```
+
+Setup Editor for Text Files
+
+```bash
+git config --global core.editor "your_editor"
+git config --list # To confirm
+```
+
+# Creating a Repo on Github
+The cloud (GitHub) serves as an online repo where files are accessible 24x7
+
+* Go to [GitHub](https://github.com/)
+
+* Create a new repo by pressing the `New Repository` button
+
+* Let us keep the name short and self explanatory and call it 'git-training' 
+
+* We will call this the `remote` repo form here on
+
+# Creating a Git Repo
+* change directory (move to another folder) using 
+```bash 
+cd
+```
+* to move back one folder use
+```bash 
+cd ..
+```
+* to make a new directory
+```bash
+mkdir
+```
+
+# Let us create a `local` Repo
+```bash
+cd ~/Documents
+# for windows users with full path
+cd C:/Users/user_name/Documents
+# create a new folder 
+# use the same name as your remote repo on GitHub
+mkdir git-training
+# move to the new folder
+cd git-training
+```
+## Convert folder to **git** repo
+```bash
+git init
+```
+## check your folder to find git configuration files/folders
+This folder is now a git repo (short for repository)
+
+# Using GitHub with Git
+If you have GitHub, you can add the repository you cloned/created to it and visualize the changes
+
+Remember, you installed this earlier?
+
+# Configuring your repo for use with GitHub on the cloud
+In order to use this local repo with the one you created earlier we need to set the remote configuration
+```bash
+git remote -v
+```
+will tell you what is the remote corresponding to your local repo
+
+If it is empty, you can setup a remote using 
+```bash
+git remote add remote_name remote_url
+```
+
+Before you do this, you need to have an account on GitHub and a repo on it
```

Comment:
That should be mentioned before, then.

Reply:
This is also mentioned before, in the initial slides

Reply:
Are these not the slides? There is no other mention of the word 'account' anywhere else.

Reply:
Ah, I thought this was a new directory; are you going to use one of the previous year's slides?

Reply:
Thank you for pointing this out. I completely missed the account creation part on the slides.